src/core/Board.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "Piece.h"
+
+#include <array>
+#include <memory>
+#include <optional>
+
+namespace core {
+
+static const size_t kRowsNumber    = 8;
```

Comment:
constexpr?

Reply:
Ok they are constexpr anyway so you can decide whether you want to declare constexpr or not.